jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JDTServicesManager.java

```diff
@@ -1,159 +1,174 @@
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4jakarta.jdt.core;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.core.JavaProject;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.PublishDiagnosticsParams;
+import org.eclipse.lsp4jakarta.commons.JakartaDiagnosticsParams;
+import org.eclipse.lsp4jakarta.commons.JakartaJavaCodeActionParams;
+import org.eclipse.lsp4jakarta.jdt.codeAction.CodeActionHandler;
+import org.eclipse.lsp4jakarta.jdt.core.annotations.AnnotationDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.beanvalidation.BeanValidationDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.cdi.ManagedBeanDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.di.DependencyInjectionDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.jax_rs.Jax_RSClassDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.jax_rs.ResourceMethodDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.jsonb.JsonbDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.jsonp.JsonpDiagnosticCollector;
+import org.eclipse.lsp4jakarta.jdt.core.persistence.PersistenceEntityDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.persistence.PersistenceMapKeyDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.servlet.FilterDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.servlet.ListenerDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.servlet.ServletDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.transactions.TransactionsDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.websocket.WebSocketDiagnosticsCollector;
+
+/**
+ * JDT manager for Java files Modified from
+ * https://github.com/eclipse/lsp4mp/blob/master/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/PropertiesManagerForJava.java
+ * with methods modified and removed to fit the purposes of the Jakarta Language
+ * Server
+ * 
+ */
+public class JDTServicesManager {
+
+    private List<DiagnosticsCollector> diagnosticsCollectors = new ArrayList<>();
+
+    private static final JDTServicesManager INSTANCE = new JDTServicesManager();
+
+    private final CodeActionHandler codeActionHandler;
+
+    public static JDTServicesManager getInstance() {
+        return INSTANCE;
+    }
+
+    private JDTServicesManager() {
+        diagnosticsCollectors.add(new ServletDiagnosticsCollector());

```

Comment:
Should we be checking if the corresponding API lies on the classpath before adding the diagnostic collector?

Reply:
What do you mean?

Reply:
For example, checking if they are using the `jakarta.servlet` specification in their project before adding the ServletDiagnosticsCollector

Reply:
Yes, but we may have to monitor any project changes, such as project classpath change, I think it's a separate enhance.

Reply:
Sounds good, let's open a follow up enhancement issue for this